broker/data-access-layer/bosh/test/bosh.BoshDirectorClient.spec.js

```diff
@@ -1325,10 +1325,14 @@ describe('bosh', () => {
 
         dummyBoshDirectorClient.makeRequestWithConfig({}, 200, directorConfig)
           .then(() => {
-            expect(requestStub).to.be.calledOnce;
+            expect(requestStub).to.be.calledTwice;
             sandbox.restore();
             config.directors = prevConfigDirectors;
             done();
+          })
+          .catch(() => {
+            sandbox.restore();
+            done(new Error('expected success but recieved error'));
           });
```

Comment:
In case the promise is rejected, we need to handle that as well because when it fails it does not restores the sandbox and other UTs also fail.